packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,304 @@
+        FROM audit_logs
+        ${whereClause}
+        ORDER BY timestamp DESC, id DESC
+        LIMIT ${limit}
+      `,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async totalAuditLogs(organizationSlug: string): Promise<number> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      throw new GraphQLError('Unauthorized: You are not authorized to perform this action');
+    }
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT count() as total
+        FROM audit_logs
+        WHERE organization_id = ${organizationSlug}
+      `,
+      queryId: 'total-audit-logs',
+      timeout: 10000,
+    });
+
+    const parsedResult = TotalAuditLogsSchema.safeParse(result.data[0]);
+    if (!parsedResult.success) {
+      throw new Error('Failed to parse audit log count result');
+    }
+
+    return Number(parsedResult.data.total);
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: string; endDate?: string },
+    pagination?: { first: number; cursorId: string | null; cursorTimestamp: string | null },
+  ): Promise<{ ok: { url: string } | null } | { error?: { message: string } | null }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      return {
+        error: {
+          message: 'Unauthorized: You are not authorized to perform this action',
+        },
+        ok: null,
+      };
+    }
+
+    const formattedStartDate = filter?.startDate ? new Date(filter.startDate) : undefined;
+    const formattedEndDate = filter?.endDate ? new Date(filter.endDate) : undefined;
+
+    try {
+      const { email } = await this.session.getViewer();
+      const totalAuditLogs = await this.totalAuditLogs(organizationSlug);
+      const getAllAuditLogs = await this.getPaginatedAuditLogs(
+        organizationSlug,
+        {
+          first: totalAuditLogs ?? 1000,
+          cursorId: pagination?.cursorId ?? null,
+          cursorTimestamp: pagination?.cursorTimestamp ?? null,
+        },
+        { startDate: formattedStartDate, endDate: formattedEndDate },
+      );
+
+      if (!getAllAuditLogs || !getAllAuditLogs.data || getAllAuditLogs.data.length === 0) {
+        return {
+          ok: null,
+          error: {
+            message: 'No audit logs found for the given organization',
+          },
+        };
+      }
+
+      const currentDate = new Date().toLocaleDateString();
+      const currentTime = new Date().toLocaleTimeString();
+      const customHeader = [
+        [`Hive Audit Logs for Organization: ${organizationSlug}`],
+        [`Date: ${currentDate}`],
+        [`Time: ${currentTime}`],
+        [`User: ${email}`],
+        [''], // Blank row
+      ];
```

Comment:
@TuvalSimha Let's omit the custom header, it will just make it more annoying for people to parse the CSV.

Reply:
Done :)